### PR TITLE
Try to add the missing "JavaFX" option, and many translations

### PR DESCRIPTION
**Pull request by QilanLin**
There is an issue I discovered after watching PPA videos of Week 5 about JavaFX. For the UI of Chinese and many other languages, the “JavaFX” option is missing in the pop-up shown when creating a new class.

I append the entry for “JavaFX” in Chinese translation, and look forwords to see if it could fix the bug.

Improved and add many new Chinese translations of “labels”

Create 4 meticulous translations for `JavaFX`, `unit test`, `enum` and `record` templates

In Chinese, the convention is to include the original English term in parentheses thereafter when it has multiple different translations or is used as a reference. It can also for the convenience of me (and other students) to master in different languages the knowledge learned at King’s.

### CODE REVIEW

**Comment by neilccbrown** [comment]
Looks good, thank you!

File: bluej/lib/chinese/labels [hunk 1]
```diff
@@ -21,19 +21,22 @@ debugger.assert.notNull=\u975e null
...
 
-debugger.inspector.title=BlueJ:  \u67e5\u770b\u5668
-debugger.inspector.object.title=BlueJ:  \u5bf9\u8c61\u67e5\u770b\u5668
-debugger.inspector.class.title=BlueJ:  \u7c7b\u67e5\u770b\u5668
-debugger.inspector.enum.title=BlueJ:  \u679a\u4e3e \u67e5\u770b\u5668
-debugger.inspector.interface.title=BlueJ:  \u63a5\u53e3 \u67e5\u770b\u5668
-debugger.inspector.result.title=BlueJ:  \u65b9\u6cd5\u8fd4\u56de\u503c
+debugger.inspector.title=BlueJ:  \u68c0\u89c6\u5668
+debugger.inspector.object.title=BlueJ:  \u5bf9\u8c61\u68c0\u89c6\u5668
+debugger.inspector.class.title=BlueJ:  \u7c7b\u68c0\u89c6\u5668
```
Comment (PwtKCL) [inline bluej/lib/chinese/labels:26]: looks like there extra spaces here, at least definitely between \ and u for \u68c0

File: bluej/lib/chinese/labels [hunk 2]
```diff
@@ -21,19 +21,22 @@ debugger.assert.notNull=\u975e null
...
+debugger.inspector.enum.title=BlueJ:  \u679a\u4e3e\u68c0\u89c6\u5668
+debugger.inspector.interface.title=BlueJ:  \u63a5\u53e3\u68c0\u89c6\u5668
+debugger.inspector.result.title=BlueJ:  \u65b9\u6cd5\u6267\u884c\u7ed3\u679c
 debugger.inspector.result.returned=\u8fd4\u56de:
-debugger.inspector.showClass=\u663e\u793a\u9759\u6001\u6210\u5458\u53d8\u91cf
-debugger.inspector.inspect=\u67e5\u770b
+debugger.inspector.object.noFields=\u65e0\u5b57\u6bb5
```
Comment (PwtKCL) [inline bluej/lib/chinese/labels:31]: Again, there might be an extra space unwanted here, but the main issue seems to be with "\u57"  (was it meant to be \u5b57 (字)

File: bluej/lib/chinese/labels [hunk 3]
```diff
@@ -21,19 +21,22 @@ debugger.assert.notNull=\u975e null
...
+debugger.inspector.class.noFields=\u65e0\u5b57\u6bb5
```
Comment (PwtKCL) [inline bluej/lib/chinese/labels:32]: Here and all others that may appear below: is the space deliberate?

File: bluej/lib/chinese/labels [hunk 4]
```diff
@@ -21,19 +21,22 @@ debugger.assert.notNull=\u975e null
...
+debugger.inspector.showClass=\u663e\u793a\u9759\u6001\u5b57\u6bb5
```
Comment (PwtKCL) [inline bluej/lib/chinese/labels:33]: It looks like there have been a mixed up between the last 2 unicode values (was it \u5b57\u6bb5 (字段)

File: bluej/lib/chinese/labels [hunk 5]
```diff
@@ -327,15 +336,23 @@ pkgmgr.test.popup.fixtureToBench=\u6d4b\u8bd5\u5de5\u5177\u5230\u5bf9\u8c61\u69f
...
 pkgmgr.newClass.stdclass=\u7c7b
 pkgmgr.newClass.interface=\u63a5\u53e3
 pkgmgr.newClass.applet=Applet
 pkgmgr.newClass.appletj=Applet
 pkgmgr.newClass.unittest=\u5355\u5143\u6d4b\u8bd5
 pkgmgr.newClass.enum=\u679a\u4e3e
+pkgmgr.newClass.JavaFXclass= JavaFX \u7c7b
+pkgmgr.newClass.record=Record
+pkgmgr.newClass.error.noType=\u672a\u9009\u62e9\u65b0\u7c7b\u7684\u7c7b\u578b
+pkgmgr.newClass.error.typeNotAvailable=\u6240\u9009\u7c7b\u578b\u5728 $LANGUAGE \u4e2d\u4e0d\u53ef\u7528
```
Comment (PwtKCL) [inline bluej/lib/chinese/labels:353]: Problem at the end of this translation: (maybe the last "u7525" should be removed?

File: bluej/lib/chinese/labels [hunk 6]
```diff
@@ -327,15 +336,23 @@ pkgmgr.test.popup.fixtureToBench=\u6d4b\u8bd5\u5de5\u5177\u5230\u5bf9\u8c61\u69f
...
+pkgmgr.newClass.error.notValidClassName=\u4e0d\u662f\u6709\u6548\u7684 $LANGUAGE \u7c7b\u540d
+pkgmgr.newClass.error.windowsRestricted=\u7c7b\u540d\u4e3a Windows \u9650\u5236\u8bcd
```
Comment (PwtKCL) [inline bluej/lib/chinese/labels:355]: Probably an escape character missing at "\u8bcdu540d" (for u540d --> \u540d, 名),
Problem at the end of this translation: (maybe the last "u8bcd should be removed?

**Review by PwtKCL — COMMENTED** [review]
Thanks again so much for the help in translating BlueJ for us.

While we are not able to semantically or grammatically check the translations, there seems to be a few things that need a correction.

The template files look fine. For the labels files however, I've used a tool (https://magictool.ai/tool/unicode-decoder-encoder/) to see the conversion between escaped to unescaped unicode characters and it seems a few errors are present: I've pointed them out in the code review comments.

Throughout the label files, I am wondering if there are some extra spaces that shouldn't be there. I can't really tell but when there is no space in the English version (in particular for a label start) then the same should apply in the translation. So double check the spaces to make they are not errors (some may be necessary, that depends on the Chinese language and therefore I won't be able to judge it).

Thank you again for your efforts!! (we need to make a better tool to handle translations editing at some point...)

**Comment by QilanLin** [comment]
Sorry for any inconvenience caused above. But all unicode characters are rendered properly on my side at https://magictool.ai/tool/unicode-decoder-encoder/ and https://checkserp.com/encode/unicode/ (The tool I was using). Could you send the screenshots for me to better locate the problems? My email is qilan.lin@kcl.ac.uk .

**Comment by PwtKCL** [comment]
As replied by mail... something I can't explained happened and the code I was looking at wasn't as it looks now. Maybe some odd thing with GitHub I don't know. Clearly the issues I had seen are not there in the source code so I've removed the code review comments. Weird.

**Pull request merged**